### PR TITLE
Added delete user and tests

### PR DESCRIPTION
**Pull request by helenli8181**
### List of Changes:

- Added `deleteUser()` function in `UserService.java`
- Created `DeleteMapping` in UserController
- Added some tests in `UserTests` and `UserServiceTests`

### Notes:

- The variable `POST_USER` could be renamed. It sounds like a verb and not only post requests use it now. Any ideas?

### CODE REVIEW

File: src/main/java/com/rocketden/main/service/UserService.java
Comment (zpChris) [inline src/main/java/com/rocketden/main/service/UserService.java:50]: Thoughts on returning error handling similar to the other functions instead of `true` / `false`?

File: src/test/java/com/rocketden/main/UserTests.java
Comment (zpChris) [inline src/test/java/com/rocketden/main/UserTests.java:150]: I know `POST_USER` is the same value as `DELETE_USER`, but I think this makes it hard to follow from a readability standpoint. I'd recommend either renaming the variable or creating a duplicate variable named `DELETE_USER`.

**Review by zpChris — CHANGES_REQUESTED** [review]
Looks good! Only left a nit and a change wondering if we should modify the return type of this, so it's not just `true` and `false` and instead provides more details.

**Comment by alankbi** [comment]
This looks good, although do we also want to include some way to remove a user from a room?

File: src/test/java/com/rocketden/main/UserTests.java
Comment (helenli8181) [reply src/test/java/com/rocketden/main/UserTests.java:150]: How does just `USER` sound?

File: src/main/java/com/rocketden/main/service/UserService.java
Comment (helenli8181) [reply src/main/java/com/rocketden/main/service/UserService.java]: Done, per our discussion yesterday!

File: src/main/java/com/rocketden/main/service/UserService.java
Comment (alankbi) [inline src/main/java/com/rocketden/main/service/UserService.java:42]: If there are multiple users with the given nickname, does this just get the first user?

File: src/main/java/com/rocketden/main/service/UserService.java
Comment (alankbi) [inline src/main/java/com/rocketden/main/service/UserService.java:48]: nit: I think you can delete this line and change the return to just be `return UserMapper.toDto(user)`

File: src/test/java/com/rocketden/main/UserTests.java
Comment (alankbi) [inline src/test/java/com/rocketden/main/UserTests.java:149]: nit: I would suggest doing something similar to the above tests, where it gets the returned objects and verifies that they match up correctly.

File: src/test/java/com/rocketden/main/UserTests.java
Comment (alankbi) [inline src/test/java/com/rocketden/main/UserTests.java:160]: Same as my other comment, I'd suggest adding a check that the returned error message is correct (see the other tests)

File: src/test/java/com/rocketden/main/service/UserServiceTests.java
Comment (alankbi) [inline src/test/java/com/rocketden/main/service/UserServiceTests.java:18]: nit: personal preference, but I think we should stick with importing each item individually (see [here](https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad)). Sometimes the IDE can auto-correct imports into the wildcard tho so usually no big deal if it slips through

File: src/test/java/com/rocketden/main/service/UserServiceTests.java
Comment (alankbi) [inline src/test/java/com/rocketden/main/service/UserServiceTests.java:47]: nit: Sorry if this is overkill, but I'd also suggest verifying in these two tests that the exception responses contain the correct error messages (see `RoomServiceTests.java` for examples)

**Review by alankbi — CHANGES_REQUESTED** [review]
The new design looks good! Sorry I kind of left a lot of nits haha, but all of them should be pretty quick fixes and then I think this should be good to merge 👍

File: src/test/java/com/rocketden/main/UserTests.java
Comment (alankbi) [inline src/test/java/com/rocketden/main/UserTests.java:153]: nit: Fix spelling (`deleteNonExistentUser`)

**Comment by zpChris** [comment]
I'll wait until @alankbi's comments are addressed until I review, since there are a decent number of nits. 👍

If I should review now instead, please just let me know! :)

File: src/main/java/com/rocketden/main/service/UserService.java
Comment (helenli8181) [reply src/main/java/com/rocketden/main/service/UserService.java:42]: Yes, since right now UserDto doesn't contain another ID. This depends on how findUserByNickname works though which I'm not super clear about.

**Review by alankbi — APPROVED** [review]
Approved!

File: src/main/java/com/rocketden/main/controller/v1/UserController.java
Comment (zpChris) [inline src/main/java/com/rocketden/main/controller/v1/UserController.java:33]: Am I correct in assuming that this code only goes through if the user is successfully deleted, and otherwise, it does not return the `HttpStatus.ACCEPTED` but instead throws an exception? If that's the case, should this return `HttpStatus.OK` based on [this article](https://briancaos.wordpress.com/2019/06/19/200-ok-vs-202-accepted-whats-the-difference/#:~:text=200%20OK%20means%20that%20the,the%20service%20will%20now%20start.)? (If so, you'll want to update the tests accordingly as well.)

File: src/main/java/com/rocketden/main/service/UserService.java
Comment (zpChris) [inline src/main/java/com/rocketden/main/service/UserService.java:44]: nit, personal preference: Add a comment above this `if` statement, similar to the `createUser(...)` method above. (No problem if you don't do this, just a personal preference.)

File: src/test/java/com/rocketden/main/UserTests.java
Comment (zpChris) [inline src/test/java/com/rocketden/main/UserTests.java:145]: I think we talked about this before, but thoughts on changing this to just `USER` or `USER_URI`? (Same for the tests below.)

File: src/test/java/com/rocketden/main/service/UserServiceTests.java
Comment (zpChris) [inline src/test/java/com/rocketden/main/service/UserServiceTests.java:48]: nit: Add comment on why this nickname is invalid (I can see us changing those details in the future, and if so, we may be confused by this method without an explanatory comment IMO).

**Review by zpChris — CHANGES_REQUESTED** [review]
Requesting changes because I just have a few nits, but otherwise looks great! I'll try to review quickly after you address these, since it shouldn't take long 👍

File: src/main/java/com/rocketden/main/controller/v1/UserController.java
Comment (helenli8181) [reply src/main/java/com/rocketden/main/controller/v1/UserController.java:33]: Yes to the first part. Sure I can switch it to an OK status; I mainly used ACCEPTED because I saw others use it with @deletemapping.

File: src/test/java/com/rocketden/main/UserTests.java
Comment (helenli8181) [reply src/test/java/com/rocketden/main/UserTests.java:145]: Whoops I forgot about this! Fixed now.

**Review by zpChris — APPROVED** [review]
LGTM! Thanks for making those changes 👍

**Comment by zpChris** [comment]
[Delete users from room and database](https://trello.com/c/861Jcxvj/113-delete-users-from-room-and-database)

**Comment by helenli8181** [comment]
[Delete users from room and database](https://trello.com/c/861Jcxvj/113-delete-users-from-room-and-database)

**Pull request merged**
